### PR TITLE
app-misc/elasticsearch: fix initscript

### PR DESCRIPTION
**Pull request by hydrapolic**
Incorporate changes from ES 5, make ES runnable via init script (currently all ES version >5 are broken - cannot be started).

https://bugs.gentoo.org/show_bug.cgi?id=598732
https://bugs.gentoo.org/show_bug.cgi?id=598944
https://bugs.gentoo.org/show_bug.cgi?id=604662

### CODE REVIEW

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Is there anything blocking an EAPI bump significantly?

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+pkg_preinst() {
+	if has_version '<app-misc/elasticsearch-2.3.2'; then
+		export UPDATE_NOTES=1
+	fi
+}
+
+pkg_setup() {
+	enewgroup ${MY_PN}
+	enewuser ${MY_PN} -1 /bin/bash /usr/share/${MY_PN} ${MY_PN}
+	esethome ${MY_PN} /usr/share/${MY_PN}
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Doesn't `enewuser` already cover this? What is our logic behind running `esethome` ?

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+src_prepare() {
+	rm -rf bin/*.{bat,exe}
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Missing `|| die` here

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	rm LICENSE.txt
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Missing `|| die` here

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+src_install() {
+	dodir /etc/${MY_PN}
+	dodir /etc/${MY_PN}/scripts
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Missing `keepdir` here perhaps?

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+	insinto /etc/${MY_PN}
+	doins config/*
+	rm -rf config
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Missing `|| die` here

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+	insinto /usr/share/${MY_PN}
+	doins -r ./*
+
+	insinto /usr/share/${MY_PN}/bin
+	doins "${FILESDIR}/elasticsearch-systemd-pre-exec"
+
+	chmod +x "${D}"/usr/share/${MY_PN}/bin/*
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Please use the `fperms` helper instead: https://devmanual.gentoo.org/function-reference/install-functions/index.html

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+	insinto /etc/${MY_PN}
+	doins config/*
+	rm -rf config
+
+	insinto /usr/share/${MY_PN}
+	doins -r ./*
+
+	insinto /usr/share/${MY_PN}/bin
+	doins "${FILESDIR}/elasticsearch-systemd-pre-exec"
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: If this is supposed to be an executable, it's better to use the `exeinto` and `doexe` helpers instead: https://devmanual.gentoo.org/function-reference/install-functions/index.html

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+	chmod +x "${D}"/usr/share/${MY_PN}/bin/*
+
+	keepdir /var/{lib,log}/${MY_PN}
+	keepdir /usr/share/${MY_PN}/plugins
+
+	insinto /usr/lib/tmpfiles.d
+	newins "${FILESDIR}/${MY_PN}.tmpfiles.d" "${MY_PN}.conf"
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: Please use the `systemd_newtmpfilesd` eclass function instead: https://devmanual.gentoo.org/eclass-reference/systemd.eclass/index.html

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+	insinto /etc/sysctl.d
+	newins "${FILESDIR}/${MY_PN}.sysctl.d" "${MY_PN}.conf"
+
+	newinitd "${FILESDIR}/elasticsearch.init7" "${MY_PN}"
+	newconfd "${FILESDIR}/${MY_PN}.conf3" "${MY_PN}"
+	systemd_newunit "${FILESDIR}"/${PN}.service5 "${PN}.service"
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: I get the following QA warning for this:

```
 * QA Notice: systemd units using /etc/conf.d detected:
 * usr/lib/systemd/system/elasticsearch.service:EnvironmentFile=-/etc/conf.d/elasticsearch
 * See: https://wiki.gentoo.org/wiki/Project:Systemd/conf.d_files
```

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+	elog "symlinking the init script:"
+	elog "ln -sf /etc/init.d/${MY_PN} /etc/init.d/${MY_PN}.instance"
+	elog
+	elog "Please make sure you put elasticsearch.yml and logging.yml"
+	elog "into the configuration directory of the instance:"
+	elog "/etc/${MY_PN}/instance"
+	elog
+	if ! [ -z ${UPDATE_NOTES} ]; then
+		elog "This update changes some configuration variables. Please review"
+		elog "/etc/conf.d/elasticsearch before restarting your services."
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: `${EROOT%/}/etc/conf.d/elasticsearch`

File: app-misc/elasticsearch/elasticsearch-5.1.1.ebuild [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+pkg_setup() {
+	enewgroup ${MY_PN}
+	enewuser ${MY_PN} -1 /bin/bash /usr/share/${MY_PN} ${MY_PN}
+	esethome ${MY_PN} /usr/share/${MY_PN}
+}
+
+src_prepare() {
+	rm -rf bin/*.{bat,exe}
+	rm LICENSE.txt
+}
```
Comment (gktrk) [inline app-misc/elasticsearch/elasticsearch-5.1.1.ebuild]: please call `default` before leaving the phase, unless it does something unwanted.

**Comment by hydrapolic** [comment]
Thanks @gktrk for the suggestions. Since this is my first time making changes to ES, I just wanted to make it work (without polishing it).

Let's make it properly then, I'll incorporate your suggestions.

**Comment by hydrapolic** [comment]
@gktrk, pushed a new version, thanks for the tips.

Regarding the systemd QA I've asked @patricklauer before, he told me this is ok. I don't have/use systemd on any machines (yet).

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/68fc026...d2ef837.Thanks!

**Comment by hydrapolic** [comment]
Thank you

**Pull request closed without merging**